src/rpc/rpc_impl.rs

```diff
@@ -75,7 +76,13 @@ impl<N: Network, E: Environment> RpcFunctions<N> for RpcContext<N, E> {
     /// Returns up to `MAXIMUM_BLOCK_REQUEST` blocks from the given `start_block_height` to `end_block_height` (inclusive).
     async fn get_blocks(&self, start_block_height: u32, end_block_height: u32) -> Result<Vec<Block<N>>, RpcError> {
         let safe_start_height = max(start_block_height, end_block_height.saturating_sub(E::MAXIMUM_BLOCK_REQUEST - 1));
-        Ok(self.ledger.get_blocks(safe_start_height, end_block_height)?)
+        let blocks: Result<Vec<Block<N>>, _> = self
+            .ledger
+            .get_blocks(safe_start_height, end_block_height)?
+            .map(|block_result| block_result)
+            .collect();
+
+        Ok(blocks?)
```

Comment:
```suggestion
        blocks
```
I think this simpler way should be possible here instead?

Reply:
or even better, to not even have a `blocks` binding at all (if the omission of `?` is possible)

Reply:
I agree that wrapping `blocks` in an `Ok` and using `?` is not so good. I can't omit the `?` because of the error type of the result that is a `RpcError` but maybe the new change is better. Let me know what you think or if you think of another solution. The change was made in [f4eccff](https://github.com/AleoHQ/snarkOS/pull/1753/commits/f4eccff4fbc5864b1ce3aeb3d519f1edf79bad88)